packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3080,18 +3081,22 @@ export class ContainerRuntime
 		address: string | undefined,
 		type: string,
 		content: any,
+		targetClientId?: string,
 	): ISignalEnvelope {
 		const newSequenceNumber = ++this._perfSignalData.signalSequenceNumber;
 		const newEnvelope: ISignalEnvelope = {
 			address,
 			clientSignalSequenceNumber: newSequenceNumber,
```

Comment:
How is `clientSignalSequenceNumber` used?
If targeted signals are happening, non-targeted clients will see a gap in the sequence. Is that okay?
I think we may need to have a little design for this support beyond just looking at a PR and making some shifts.

Reply:
We only track metrics for signals sent by the local client. So `clientSequenceNumber` is a sequential ID assigned to signals sent by the local client and this is compared to `trackingSignalSequenceNumber` once received (and assigned to `clientSequenceNumber` on submit).

As far as the gap seen by non-targeted clients, I don't think there is one since each client has tracks their own client-specific`clientSignalSequenceNumber`. This information is only to get information about the sending client -- kind of like the way `clientSequenceNumber` is for ops.

Reply:
I think you are saying that nothing else reads `clientSignalSequenceNumber` apart from local client. Are there other code parts that have access to `clientSignalSequenceNumber`?
We should be considering how this property is useful if there is an exposure. If there isn't, then we could change the use and not include it or not(?) bump it for non-broadcasts.

Reply:
Yes, nothing else reads `clientSignalSequenceNumber` apart from local client -- it's only current use is for local client to track lost signals.

If we're not tracking lost targeted signals, then I agree it would make sense to not increment. It won't really change much logic-wise since when a client submits a signal, `trackingSignalSequenceNumber` is only set to expect the value of `clientSignalSequenceNumber` when the signal is expected to return (broadcast signals).